src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -4855,4 +4855,18 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
 
         fn main() {}
     """)
+
+    fun `test recursive async function E0733`() = checkErrors("""
+        async fn foo() {
+            foo().<error descr="Recursion in an `async fn` requires boxing [E0733]">await</error>;
+            foo();
+            bar().await;
+        }
+        async fn bar() {}
+
+        #[async_recursion]
```

Comment:
@dima74 WDYT about adding a quick fix? As I see, the compiler is suggesting this as a fix: `note: consider using the async_recursion crate: https://crates.io/crates/async_recursion`

Reply:
Added a quick fix in case if current crate has `async_recursion` dependency

Reply:
Maybe also provide the quick-fix tests?

Reply:
Added the test